numpy/core/src/multiarray/textreading/tokenize.cpp

```diff
@@ -1,20 +1,16 @@
 
 #include <Python.h>
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <stdbool.h>
-#include <string.h>
-
 #define NPY_NO_DEPRECATED_API NPY_API_VERSION
 #define _MULTIARRAYMODULE
 #include "numpy/ndarraytypes.h"
 
-#include "textreading/stream.h"
-#include "textreading/tokenize.h"
-#include "textreading/parser_config.h"
-#include "textreading/growth.h"
-
+extern "C"{
```

Comment:
It's probably better to put the `extern "C"` in the includes instead of around them, because

1. it avoids doing so at each C++ include site
2. the way you do it propagates to all included files, including potential C++ files (guarded by `#ifdef __cplusplus`)

Reply:
OK, moved it back to the headers.  I guess we effectively should soon have a habit of just adding those to most headers.